plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/S3SelectTestHelper.java

```diff
@@ -0,0 +1,233 @@
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.plugin.hive.HiveLocationService;
+import io.trino.plugin.hive.HiveMetadataFactory;
+import io.trino.plugin.hive.HivePageSourceProvider;
+import io.trino.plugin.hive.HivePartitionManager;
+import io.trino.plugin.hive.HiveSplitManager;
+import io.trino.plugin.hive.HiveTransactionManager;
+import io.trino.plugin.hive.LocationService;
+import io.trino.plugin.hive.NamenodeStats;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.hive.NoneHiveRedirectionsProvider;
+import io.trino.plugin.hive.PartitionUpdate;
+import io.trino.plugin.hive.PartitionsSystemTableProvider;
+import io.trino.plugin.hive.PropertiesSystemTableProvider;
+import io.trino.plugin.hive.aws.athena.PartitionProjectionService;
+import io.trino.plugin.hive.fs.FileSystemDirectoryLister;
+import io.trino.plugin.hive.metastore.HiveMetastoreConfig;
+import io.trino.plugin.hive.metastore.HiveMetastoreFactory;
+import io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore;
+import io.trino.plugin.hive.s3.HiveS3Config;
+import io.trino.plugin.hive.s3.TrinoS3ConfigurationInitializer;
+import io.trino.plugin.hive.security.SqlStandardAccessControlMetadata;
+import io.trino.spi.connector.ConnectorPageSourceProvider;
+import io.trino.spi.connector.ConnectorSplitManager;
+import io.trino.spi.type.TestingTypeManager;
+import org.apache.hadoop.fs.Path;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.util.concurrent.MoreExecutors.newDirectExecutorService;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.trino.plugin.hive.HiveTestUtils.getDefaultHivePageSourceFactories;
+import static io.trino.plugin.hive.HiveTestUtils.getDefaultHiveRecordCursorProviders;
+import static io.trino.plugin.hive.TestingThriftHiveMetastoreBuilder.testingThriftHiveMetastoreBuilder;
+import static io.trino.spi.connector.MetadataProvider.NOOP_METADATA_PROVIDER;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static java.lang.String.format;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static org.testng.util.Strings.isNullOrEmpty;
+
+public class S3SelectTestHelper
+{
+    private HdfsEnvironment hdfsEnvironment;
+    private LocationService locationService;
+    private TestingHiveMetastore metastoreClient;
+    private HiveMetadataFactory metadataFactory;
+    private HiveTransactionManager transactionManager;
+    private ConnectorSplitManager splitManager;
+    private ConnectorPageSourceProvider pageSourceProvider;
+
+    private ExecutorService executorService;
+    private HiveConfig hiveConfig;
+    private ScheduledExecutorService heartbeatService;
+
+    public S3SelectTestHelper(String host,
```

Comment:
I find this class rather peculiar.
Its only purpose is to hold objects which is initialising in the constructor to tear them afterwards down (if the programmer remembers about it).

I would advocate for smaller createXXX methods where the programmer is creating the needed instances when needed. The `executorService` and `heartBeatService` should be provided IMO from the client code so that they can be eventually torn down / reused from within the caller context.

Reply:
All of these objects are needed to be created to test without extending the AbstractTest* classes for S3. This is basically emulating the setup done by the abstract classes, but avoiding changes on the same fields by multiple tests, to allow safely parallel run. Not extending the AbstractTest* and overriding the config was a change suggested by @arhimondr. 
In general, I agree with more modularity. But in this scenario, the class is designed to wrap the setup needed by a Select test, to make initialization for it easier, rather than duplicating it in JSON & scan range tests, as we only need variations to the HiveConfig.

Reply:
@findinpath Yeah, originally it was implemented by extending a base test class. But some of the settings have to be changed on per test basis. This is why a helper was added.

Reply:
The `S3SelectTestHelper` does fit the purpose of reusing code to be used in

```
readTable(tableJson,
                        s3SelectTestHelper.getTransactionManager(),
                        s3SelectTestHelper.getHiveConfig(),
                        s3SelectTestHelper.getPageSourceProvider(),
                        s3SelectTestHelper.getSplitManager())
```

What about having:
- #createTransactionManager
- #createPageSourceProvider
- #createSplitManager

? 

There is still code reuse this way as well.
Consider this optional.

Reply:
The creation of:
- transactionManager - requires a metadataFactory - this depends on hiveConfig, hdfsEnvironment, heartbeatService, locationService
- pageSourceProvider - uses hdfsEnvironment, hiveConfig
- splitManager - needs transactionManager, hdfsEnvironment, executorService, hiveConfig
I understand your reasoning, but I do not see a benefit in adding those methods for the purpose of this PR, where this wrapper is used just in our tests. Furthermore, it leaves room for errors, given the dependency of splitManager to transactionManager. I will keep this suggestion in mind for future changes. Thank you!